src/Driver/Internal/AbstractHttpDriver.php

```diff
@@ -80,67 +80,26 @@ final protected function handleRequest(Request $request): void
     }
 
     /**
-     * Write the given response to the client using the write callback provided to `setup()`.
+     * Write the given response to the client.
      */
     abstract protected function write(Request $request, Response $response): void;
 
     /**
      * Used if an exception is thrown from a request handler.
+     *
+     * This is not designed to be a general-purpose exception handler, rather a last-resort to write to the logger
+     * if the application has failed to handle an exception thrown from a {@see RequestHandler}. Instead of relying on
+     * this handler, use {@see ExceptionHandlerMiddleware} and {@see ExceptionHandler}.
      */
     private function handleInternalServerError(Request $request, \Throwable $exception): Response
     {
-        $status = HttpStatus::INTERNAL_SERVER_ERROR;
-
-        $client = $request->getClient();
-        $method = $request->getMethod();
-        $uri = (string) $request->getUri();
-        $protocolVersion = $request->getProtocolVersion();
-        $local = $client->getLocalAddress()->toString();
-        $remote = $client->getRemoteAddress()->toString();
-
-        $this->logger->error(
-            \sprintf(
-                "Unexpected %s with message '%s' thrown from %s:%d when handling request: %s %s HTTP/%s %s on %s",
-                $exception::class,
-                $exception->getMessage(),
-                $exception->getFile(),
-                $exception->getLine(),
-                $method,
-                $uri,
-                $protocolVersion,
-                $remote,
-                $local,
-            ),
-            [
-                'exception' => $exception,
-                'method' => $method,
-                'uri' => $uri,
-                'protocolVersion' => $protocolVersion,
-                'local' => $local,
-                'remote' => $remote,
-            ],
-        );
-
-        return $this->handleError($status, null, $request);
+        $this->exceptionHandler ??= new DefaultExceptionHandler($this->errorHandler, $this->logger);
```

Comment:
Is it really worth making this nullable instead of eagerly initialized?

Reply:
No, probably not. Changed.